### PR TITLE
Mono8 support

### PR DESCRIPTION
**Pull request by jxjshaw**
Image panel support for mono8, details can be found in this convo: https://github.com/cruise-automation/webviz/issues/161#issuecomment-506621910

mono88

mono888887

source of sample bag: https://www.cse.unr.edu/~kalexis/datasets/thermal-inertial/

Btw encountered some weirdness with /packages/@cruise-automation not being found. Assuming it was an issue with my fork/local npm but flagging just in case

### CODE REVIEW

File: packages/webviz-core/src/panels/ImageView/decodings.js [hunk 1]
```diff
@@ -114,3 +114,16 @@ export function decodeRGGB(rggb: Uint8Array, width: number, height: number, outp
     }
   }
 }
+
+export function decodeMono8(mono8: Uint8Array, width: number, height: number, output: Uint8ClampedArray) {
+  let inIdx = 0;
+  let outIdx = 0;
+
+  for (let i = 0; i < width * height; i++) {
+    const ch = mono8[inIdx++];
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/decodings.js]: For `16UC1` / `mono16`, I think you would want `(data[inIdx++] << 8) | data[inIdx++]` (assuming big endian – not sure where this is documented). There should be twice as many values in the input data because it's 2 bytes per value instead of 1.

File: packages/webviz-core/src/panels/ImageView/ImageCanvas.js [hunk 2]
```diff
@@ -89,6 +89,9 @@ export default class ImageCanvas extends React.Component<Props, State> {
           case "bgr8": decodeBGR(rawData, width, height, image.data); break;
           case "32FC1": decodeFloat1c(rawData, width, height, image.data); break;
           case "bayer_rggb8": decodeRGGB(rawData, width, height, image.data); break;
+          case "mono8": decodeMono8(rawData, width, height, image.data); break;
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/ImageCanvas.js]: nit: can make it clear these two are treated the same by using a fallthrough:
```suggestion
          case "mono8":
```

File: packages/webviz-core/src/panels/ImageView/ImageCanvas.js [hunk 3]
```diff
@@ -89,6 +89,9 @@ export default class ImageCanvas extends React.Component<Props, State> {
...
+          case "8UC1": decodeMono8(rawData, width, height, image.data); break;
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/ImageCanvas.js]:
```suggestion
          case "8UC1":
            decodeMono8(rawData, width, height, image.data); break;
```

File: packages/webviz-core/src/panels/ImageView/decodings.js (hunk 1 above)
Comment (jtbandes) [reply packages/webviz-core/src/panels/ImageView/decodings.js]: actually, just noticed there's an `is_bigendian` on the image: http://docs.ros.org/melodic/api/sensor_msgs/html/msg/Image.html

Comment (jtbandes) [reply packages/webviz-core/src/panels/ImageView/decodings.js]: you could also use DataView like we do for 32FC1

Comment (jxjshaw) [reply packages/webviz-core/src/panels/ImageView/decodings.js]: does that mean we should support is_bigendian = false for all the encodings?

Comment (jtbandes) [reply packages/webviz-core/src/panels/ImageView/decodings.js]: Yeah probably that would be ideal, for example this `true` should be `!is_bigendian`: https://github.com/cruise-automation/webviz/blob/f1d751aa8320e33f4a501a2e4b3a37851423a268/packages/webviz-core/src/panels/ImageView/decodings.js#L59

it doesn't affect the 8-bit encodings though.

File: packages/webviz-core/src/panels/ImageView/ImageCanvas.js [hunk 4]
```diff
@@ -81,15 +81,22 @@ export default class ImageCanvas extends React.Component<Props, State> {
...
       // Binary message processing
       if (msg.datatype === "sensor_msgs/Image") {
-        const { width, height, encoding } = msg.message;
+        const { width, height, is_bigendian, encoding } = msg.message;
         image = new ImageData(width, height);
         // prettier-ignore
         switch (encoding) {
           case "yuv422": decodeYUV(rawData, width, height, image.data); break;
-          case "bgr8": decodeBGR(rawData, width, height, image.data); break;
+          case "bgr8": decodeBGR(rawData, width, height, is_bigendian, image.data); break;
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/ImageCanvas.js]: not needed for bgr8 since it's only 8-bit

File: packages/webviz-core/src/panels/ImageView/decodings.js [hunk 5]
```diff
@@ -51,12 +51,18 @@ export function decodeBGR(bgr: Uint8Array, width: number, height: number, output
...
+  height: number,
+  is_bigendian: boolean,
+  output: Uint8ClampedArray
+) {
   const view = new DataView(gray.buffer, gray.byteOffset);
 
   let outIdx = 0;
   for (let i = 0; i < width * height * 4; i += 4) {
-    const val = view.getFloat32(i, true) * 255;
+    const val = view.getFloat32(i, !is_bigendian ? true : false) * 255;
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/decodings.js]:
```suggestion
    const val = view.getFloat32(i, !is_bigendian) * 255;
```

File: packages/webviz-core/src/panels/ImageView/decodings.js [hunk 6]
```diff
@@ -114,3 +120,36 @@ export function decodeRGGB(rggb: Uint8Array, width: number, height: number, outp
...
+  width: number,
+  height: number,
+  is_bigendian: boolean,
+  output: Uint8ClampedArray
+) {
+  let inIdx = 0;
+  let outIdx = 0;
+
+  for (let i = 0; i < width * height; i++) {
+    const val = !is_bigendian ? mono16[inIdx] : mono16[inIdx - 1];
```
Comment (jtbandes) [inline packages/webviz-core/src/panels/ImageView/decodings.js]: Did you try `... : mono16[inIdx + 1]` instead? I think `inIdx - 1` is wrong because at the beginning, it will be `mono16[-1]`.

File: packages/webviz-core/package-lock.json [hunk 7]
```diff
@@ -15,27 +15,27 @@
         "regenerator-runtime": {
```
Comment (jtbandes) [inline packages/webviz-core/package-lock.json]: Would be nice to revert these changes since they aren't relevant to the PR.

**Comment by jtbandes** [comment]
Merging PR to internal repo, will be out to open source repo in a future push
